phoenicis-apps/src/main/java/com/playonlinux/apps/ConfigurableApplicationSource.java

```diff
@@ -0,0 +1,51 @@
+package com.playonlinux.apps;
+
+import com.playonlinux.apps.dto.CategoryDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ConfigurableApplicationSource implements ApplicationsSource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurableApplicationSource.class);
+
+    private final String sourceUrl;
+    private final LocalApplicationsSource.Factory localApplicationsSourceFactory;
+
+    public ConfigurableApplicationSource(String sourceUrl, LocalApplicationsSource.Factory localApplicationsSourceFactory) {
+        this.sourceUrl = sourceUrl;
+        this.localApplicationsSourceFactory = localApplicationsSourceFactory;
+    }
+
+    @Override
+    public List<CategoryDTO> fetchInstallableApplications() {
```

Comment:
The MultipleApplicationSources will be created every time this is called. It would be better to create the MultipleApplicationSources only once in the constructor.